stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/state/mediagallerypreview/MediaGalleryPreviewActivityAttachmentState.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.compose.state.mediagallerypreview
+
+import android.os.Parcelable
+import io.getstream.chat.android.client.models.Attachment
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Class used to transform [Attachment] into a smaller and easily
+ * parcelable version that contains the minimum necessary data
+ * for the proper functioning of the Media Gallery Preview screen.
+ *
+ * @param name The name of the attachment.
+ * @param url The url of the file.
+ * @param thumbUrl The URL for the thumbnail version of the attachment,
+ *  given the attachment has a visual quality, e.g. is a video, an image,
+ *  a link to a website or similar.
+ * @param imageUrl The URL for the raw version of the attachment.
+ * @param assetUrl The URL for the asset.
```

Comment:
If we could differentiate these two or link to client side docs, it would be good, as I have no idea what this means on the face of it

Reply:
Added a finer explanation here.

Our Attachment properties are sometimes confusing and difficult to grasp/ explain, especially since we sometimes have discrepancies between FE SDKs.